projects/ad2s1210_iio/app/ad2s1210_iio.c

```diff
@@ -0,0 +1,619 @@
+ * @param	chn_mask[in] - Channels select mask
+ * @return	0 in case of success, negative error code otherwise
+ */
+static int32_t iio_ad2s1210_prepare_transfer(void *dev_instance,
+		uint32_t chn_mask)
+{
+#if (DATA_CAPTURE_MODE == CONTINUOUS_DATA_CAPTURE)
+
+	int32_t ret;
+
+	ret = iio_trig_enable(ad2s1210_hw_trig_desc);
+	if (ret)
+		return ret;
+#endif
+	return 0;
+}
+
+/**
+ * @brief	Perform tasks before end of current data transfer
+ * @param	dev[in] - IIO device instance
+ * @return	0 in case of success, negative error code otherwise
+ */
+static int32_t iio_ad2s1210_end_transfer(void *dev)
+{
+#if (DATA_CAPTURE_MODE == CONTINUOUS_DATA_CAPTURE)
+	int32_t ret;
+
+	ret = iio_trig_disable(ad2s1210_hw_trig_desc);
+	if (ret)
+		return ret;
+#endif
+	return 0;
+}
+
+int32_t ad2s1210_trigger_handler(struct iio_device_data *iio_dev_data)
+{
+	enum ad2s1210_channel chn;
+	int32_t ret;
+	uint16_t data;
+	uint8_t i;
+
+	if (!buf_size_updated) {
+		/* Update total buffer size according to bytes per scan for
+		 * proper alignment of multi-channel IIO buffer data
+		 */
+		iio_dev_data->buffer->buf->size = ((uint32_t)(DATA_BUFFER_SIZE /
+						   iio_dev_data->buffer->bytes_per_scan))
+						  * iio_dev_data->buffer->bytes_per_scan;
+		buf_size_updated = true;
+	}
+
+	/* Push into circular buffer */
+	for (i = 0; i < RESOLVER_CHANNELS; i++) {
+		if (ad2s1210_iio_channels[i].ch_type == IIO_ANGL)
+			chn = AD2S1210_POS;
+		else
+			chn = AD2S1210_VEL;
+
+		if (!(iio_dev_data->buffer->active_mask & (1 << i)))
+			continue;
+
+		ret = ad2s1210_spi_single_conversion(ad2s1210_dev_inst,
+						     chn, &data);
+		if (ret)
+			return ret;
+
+		ret = no_os_cb_write(iio_dev_data->buffer->buf, &data,
+				     BYTES_PER_SAMPLE);
+		if (ret)
+			return ret;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief	Init for reading/writing and parameterization of a
+ *			ad2s1210 IIO device
+ * @param	desc[in,out] - IIO device descriptor
+ * @return	0 in case of success, negative error code otherwise
+ */
+static int32_t ad2s1210_iio_param_init(struct iio_device **desc)
+{
+	struct iio_device *iio_ad2s1210_inst;
+
+	iio_ad2s1210_inst = calloc(1, sizeof(struct iio_device));
+	if (!iio_ad2s1210_inst)
+		return -ENOMEM;
+
+	iio_ad2s1210_inst->num_ch = NO_OS_ARRAY_SIZE(ad2s1210_iio_channels);
+	iio_ad2s1210_inst->channels = ad2s1210_iio_channels;
+	iio_ad2s1210_inst->attributes = ad2s1210_iio_global_attributes;
+
+	iio_ad2s1210_inst->submit = iio_ad2s1210_submit_buffer;
+	iio_ad2s1210_inst->pre_enable = iio_ad2s1210_prepare_transfer;
+	iio_ad2s1210_inst->post_disable = iio_ad2s1210_end_transfer;
+	iio_ad2s1210_inst->debug_reg_read = iio_ad2s1210_debug_reg_read;
+	iio_ad2s1210_inst->debug_reg_write = iio_ad2s1210_debug_reg_write;
+#if (DATA_CAPTURE_MODE == CONTINUOUS_DATA_CAPTURE)
+	iio_ad2s1210_inst->trigger_handler = ad2s1210_trigger_handler;
+#endif
+
+	*desc = iio_ad2s1210_inst;
+
+	return 0;
+}
+
+#if (DATA_CAPTURE_MODE == CONTINUOUS_DATA_CAPTURE)
```

Comment:
This check is not really needed.

Reply:
its to avoid compiler warinings in case BURST data capture is defined
[Warning] ad2s1210_iio.c@494,16: 'ad2s1210_iio_trigger_param_init' defined but not used [-Wunused-function]

Reply:
Compiler warnings should not be an issue. This check is not really added in the other projects. hence the suggestion